src/particle/pdgid/functions.py

```diff
@@ -129,13 +129,16 @@ def is_meson(pdgid):
         return False
     if abspid(pdgid) <= 100:
         return False
-    if _fundamental_id(pdgid) <= 100 and _fundamental_id(pdgid) > 0:
+    if 0 < _fundamental_id(pdgid) <= 100:  # type: ignore
```

Comment:
```suggestion
    if 0 < int(_fundamental_id(pdgid)) <= 100:
```

This will make mypy happy, and not drop type checking for these lines. SupportsInt doesn't require that comparisons be defined, which is why this is complaining.

Reply:
Technically, this is a valid input to any of these functions:

```python
class MyVal:
    def __int__(self):
        return 3
```

But these functions with the comparisons will break if you don't call int() first, which is why mypy is (correctly) complaining.